v2/tools/generator/internal/functions/get_extended_resources_function.go

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package functions
+
+import (
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astbuilder"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/dave/dst"
+	"sort"
+)
+
+const (
+	ExtendedResourcesFunctionName = "GetExtendedResources"
+)
+
+type GetExtendedResourcesFunction struct {
+	idFactory  astmodel.IdentifierFactory
+	resources  []astmodel.TypeName
+	packageRef *astmodel.PackageReferenceSet
+}
+
+// Ensure GetExtendedResources properly implements Function
+var _ astmodel.Function = &GetExtendedResourcesFunction{}
+
+// NewGetExtendedResourcesFunction creates a new GetExtendedResources
+func NewGetExtendedResourcesFunction(idFactory astmodel.IdentifierFactory, resources []astmodel.TypeName, references []astmodel.PackageReference) *GetExtendedResourcesFunction {
```

Comment:
As I mentioned elsewhere, I think you can build the []`astmodel.PackageReference` from the `[]astmodel.TypeName`'s in this case, so I don't think you need both

Reply:
I'd suggest removing `references` from the factory func, and you don't need to keep it on the struct either - just create the `PackageReferenceSet` on the fly inside `RequiredPackageReferences()`.